packages/date-time-picker/src/DateTimePicker.ts

```diff
@@ -0,0 +1,158 @@
+
+import '@spectrum-web-components/calendar/sp-calendar.js';
+import '@spectrum-web-components/field-label/sp-field-label.js';
+import '@spectrum-web-components/icons-workflow/icons/sp-icon-calendar.js';
+import '@spectrum-web-components/overlay/sp-overlay.js';
+import '@spectrum-web-components/picker-button/sp-picker-button.js';
+import '@spectrum-web-components/popover/sp-popover.js';
+
+/**
+ * @element sp-date-time-picker
+ *
+ * @event change - Announces when a new date/time is defined by emitting a `Date` object
+ *
+ * @slot calendar-icon - The icon used in the calendar button
+ * @slot help-text - Default or non-negative help text to associate to your form element
+ * @slot negative-help-text - Negative help text to associate to your form element when `invalid`
+ */
+export class DateTimePicker extends InputSegments {
+    public static override get styles(): CSSResultArray {
+        return [...super.styles, styles];
+    }
+
+    /**
+     * Indicates whether the picker should be displayed or not
+     */
+    @property({ type: Boolean, reflect: true })
+    public open = false;
+
+    @state()
+    override includeDate = true;
+
+    @state()
+    override includeTime = true;
+
+    @state()
+    private pickerDate?: Date;
+
+    protected override renderField(): TemplateResult {
+        return html`
+            ${super.renderInputContent()} ${this.renderPicker()}
+        `;
+    }
+
+    public renderPicker(): TemplateResult {
+        const isDisabled = super.disabled || super.readonly;
+
+        return html`
+            <div class="picker">
+                ${super.renderStateIcons()}
+
+                <sp-picker-button
+                    ?open=${this.open}
+                    ?quiet=${super.quiet}
+                    ?invalid=${super.invalid}
+                    ?disabled=${isDisabled}
+                    @click=${this.openPicker}
+                >
+                    <slot name="calendar-icon" slot="icon">
+                        <sp-icon-calendar></sp-icon-calendar>
+                    </slot>
+                </sp-picker-button>
+
+                <!-- TODO: Enable "receives-focus" when calendar is navigable via keyboard -->
+                <sp-overlay
+                    placement="bottom-start"
+                    receives-focus="false"
+                    .triggerElement=${this as HTMLElement}
+                    ?open=${this.open}
+                    @sp-closed=${this.closePicker}
+                >
+                    <sp-popover class="popover">
+                        <div class="popover-content">
+                            <sp-calendar
+                                .selectedDate=${this.pickerDate}
+                                @change=${this.handleDate}
+                            ></sp-calendar>
+                        </div>
+                    </sp-popover>
+                </sp-overlay>
+            </div>
+        `;
+    }
+
+    public openPicker(): void {
```

Comment:
Can we use `showPicker()` to normalize to the platform API: https://developer.mozilla.org/en-US/docs/Web/API/HTMLInputElement/showPicker

This would imply `hidePicker()` below.